api/bazel/api_build_system.bzl

```diff
@@ -74,6 +74,17 @@ def api_go_grpc_library(name, proto, deps = []):
         ]
     )
 
+# This is api_proto_library plus some logic internal to //envoy/api.
+def api_proto_library_internal(name, visibility = ["//visibility:private"], srcs = [], deps = [], has_services = 0, require_py = 1):
```

Comment:
You can use `**kwargs` to pass-thru args that aren't explicitly used in the method body (it's basically like Python).

Reply:
Done.  Please take another look.